packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:aws_signature_v4/aws_signature_v4.dart';
+
+import 'impl/analytics_client/analytics_client.dart';
+import 'impl/analytics_client/shared_prefs.dart';
+import 'impl/device_context_info_provider.dart';
+
+// TODO - what are campaignattributes android? (setCampaignAttributes)
+
+/// Public facing Plugin
+/// Validate and parse inputs
+/// Before delegating to AnalyticsClient
+class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
+  late AnalyticsClient? _analyticsClient;
```

Comment:
Why does this need to be a later initializer? Why does it need to be nullable?

Reply:
late => because it is only initialized in configure call.

nullable => in case user forgets to call configure, need way to check if analyticsClient is not enabled.  I have just updated code in this class to check nullable value and throw exception otherwise.